src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt

```diff
@@ -29,11 +30,28 @@ class GroupService(
             contentType = contentType,
             sizeInBytes = size,
             file = Binary(data),
-            attachmentHistory = listOf(createNewAttachmentHistory(userId, size)),
+            attachmentHistory = listOf(createNewAttachmentHistory(userId, size, contentType)),
         )
         return groupAttachmentRepository.save(groupAttachment)
     }
 
+    fun updateAttachment(data: ByteArray, attachmentId: String, groupId: String, userId: String): GroupAttachment {
```

Comment:
shouldn't we allow only Attachment author to change it? If i am not mistaken we've got only two possible scenarios for adding new group attachment: groupImage & expenseAttachment. I think we should allow only group creator to change image (as only group creator can change group name). Same case for expense attachment.

Reply:
Will reports be stored in attachment store? If yes, i don't think we should allow to change them.

Reply:
what if somebody would try to update image with for example csv report

Reply:
will discuss on daily